compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -3421,14 +3425,20 @@ impl<'a> Parser<'a> {
         pth: ast::Path,
         recover: bool,
         close_delim: Delimiter,
-    ) -> PResult<'a, (ThinVec<ExprField>, ast::StructRest, bool)> {
+    ) -> PResult<
+        'a,
+        (
+            ThinVec<ExprField>,
+            ast::StructRest,
+            Result<(), ErrorGuaranteed>, /* aync blocks are forbidden in Rust 2015 */
+        ),
+    > {
```

Comment:
I haven't reviewed this function yet as the changes look non-trivial and I'm a bit exhausted now.
**Note to self**: Review this later.